### ISSUE TITLE
iOS double notification while app is in the background 

### ISSUE THREAD

**Issue by nikolajev-dev** [dpa99c/cordova-plugin-firebasex#158]
# Bug report

**Current behavior:**
While app is in the background and foreground_notification is set to true, receive 2 notifications in status bar instead of one. This is happening only when push notification using APNS token. When push notification with firebase token, with same payload, all works fine.

Receiving 2 notification instead of one while app is in the background

I tested on exaple project, it act the same as on my project

**Expected behavior:**

Receive one notification while app is running in the background and when receive notification from APNS with foreground_notification key set.
**Steps to reproduce:**
1. Run app 
2. Get tocken usnig method getApnsToken
3. Use apns token to apns push notification

**Environment information**
- Cordova CLI version 
- Cordova platform version
	- cordova platform ls
- Plugins & versions installed in project (including this plugin)
    - cordova plugin ls 6.1.0
- Dev machine OS and version, e.g.
    - OSX 10.14.4
        
_Runtime issue_
- OS details
    - iOS 12.1,
	
_iOS build issue:_
- Node JS version
    10.15.3
- XCode version
    10.1

**Related code:**
```
{
    "aps":{
        "alert":{
            "body":"Test",
            "title":"Test"
        },
        "content-available":1,
        "category":"NEW_MESSAGE_CATEGORY",
        "badge":2,
        "mutable-content":1,
        "sound":"default"
        },  
    "notification_foreground": true
}
```

**Comment by odai-alali** [comment]
I am having the same problem, were you able to find a solution/workaround?

**Comment by nikolajev-dev** [comment]
Not yet. I hope plugin developers will come up with some solutions.

**Comment by dpa99c** [comment]
This is caused because you have both `"notification_foreground": "true"` and `"content-available":1` in the message payload.

`"notification_foreground": "true"` causes the [`willPresentNotification` delegate](https://github.com/dpa99c/cordova-plugin-firebasex/blob/master/src/ios/AppDelegate%2BFirebasePlugin.m#L295) to trigger a notification.
`"content-available":1` causes the [`didReceiveRemoteNotification` delegate](https://github.com/dpa99c/cordova-plugin-firebasex/blob/master/src/ios/AppDelegate%2BFirebasePlugin.m#L141) to trigger a notification.

TL;DR: Don't use `"notification_foreground": "true"` and `"content-available":1` together in the same message payload.
I will update the docs to make this clear.

**Comment by dpa99c** [comment]
After some investigation, I've been able to handle the case where both both `"notification_foreground": "true"` and `"content-available":1` are specified in the message payload. 
The above commit handles this to ensure that only one notification is created per message received.

This means that you can now use `"content-available":1` to have iOS wake up the app while it's in the background to receive the message payload without user interaction and still have a notification displayed in the message arrives while the app is in the foreground by using `"notification_foreground": "true"` .

This functionality will be merged to master and published in the next released version to npm.
